app/controllers/admin/ministries_controller.rb

```diff
@@ -0,0 +1,62 @@
+class Admin::MinistriesController < ApplicationController
+	layout 'admin_panel_sidenav'
+  before_action :authenticate_user!
+	before_action :require_admin, only: [:index, :update, :edit, :show, :approve, :reject, :destroy]
+	before_action :set_ministry, only: [:edit, :update, :show, :approve, :reject, :destroy]
+
+	def index
+    @ministries = Ministry.all.includes(:created_by).order(created_at: :desc).filter_by(params[:page], filter_params.to_h, sort_params.to_h)
+    respond_to do |format|
+      if request.xhr?
+        format.html {render partial: 'admin/ministries/table', locals: { ministries: @ministries } }
+      else
+        format.html
+      end
+    end
+	end
+
+	def show
+	end
+
+	def update
+		if @ministry.update_attributes(secure_params)
+			redirect_back fallback_location: root_path, flash_success_info: 'Ministry details was successfully updated.'
+		else
+			redirect_back fallback_location: root_path, flash_info: 'Ministry details was not successfully updated.'
+		end
+	end
+
+	def destroy
+		@ministry.destroy
+		redirect_to admin_ministries_path, flash_success_info: 'Ministry was successfully deleted.'
+	end
+
+	def approve
```

Comment:
@BALAJI-BALA - this should be in the model, also you should store who is approving the ministry in meta